core/codegen/src/derive/form_field.rs

```diff
@@ -300,30 +300,21 @@ pub fn default<'v>(
 ) -> Result<Option<syn::Expr>> {
     let mut exprs = FieldAttr::from_attrs(FieldAttr::NAME, &field.attrs)?
         .into_iter()
-        .map(|a| a.default)
-        .flat_map(move |expr| {
-            let field_span = field.ident().span()
-                .join(field.ty.span())
-                .unwrap_or(field.ty.span());
-
-            expr.map(|expr| {
-                let span = expr.key_span.unwrap_or(field_span);
-                define_spanned_export!(span => _form);
-                syn::parse2(quote_spanned!(span => {
-                    let __default: Option<_> = { #expr }.into();
-                    __default
-                })).unwrap()
-            })
+        .filter_map(|a| a.default)
+        .map(move |expr| {
+            let ty = &field.ty;
+            define_spanned_export!(expr.span => _form);
+            syn::parse2(quote_spanned!(expr.span => {
+                let __default: Option<#ty> = { #expr }.into();
```

Comment:
So there is a problem with this line. The `into` is used now to convert expr into an Option, so it only works if `#expr` is already of the type `#ty`. We wanted to have the into conversion, so it think this should be `let __default: Option<_> = Some({ #expr }.into());`